backend/src/main/java/com/google/sps/notifs/EmailNotifier.java

```diff
@@ -0,0 +1,174 @@
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+/** Class representing the notification system capable of sending mails to the user. */
+public class EmailNotifier {
+
+  private static final String APPLICATION_NAME = "Ad-lib";
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String TOKENS_DIRECTORY_PATH = "tokens";
+  /**
+   * Global instance of the scopes required by this quickstart. If modifying these scopes, delete
+   * your previously saved tokens/ folder.
+   */
+  private static final List<String> SCOPES =
+          Collections.singletonList(GmailScopes.MAIL_GOOGLE_COM);
+
+  private static final String CREDENTIALS_FILE_PATH = "credentials.json";
+
+  /** Variable representing the email to be notified */
+  private final String toEmail;
+
+  private final String recipientName;
+
+  /**
+   * @param recipientName Name of the recipient of the user
+   * @param toEmail Email that this email is going to sent to.
+   */
+  public EmailNotifier(String recipientName, String toEmail) {
+    this.recipientName = recipientName;
+    this.toEmail = toEmail;
+  }
+
+  /**
+   * Creates an authorized Credential object.
+   *
+   * @param HTTP_TRANSPORT The network HTTP Transport.
+   * @return An authorized Credential object.
+   * @throws IOException If the credentials.json file cannot be found.
+   */
+  private static Credential getCredentials(final NetHttpTransport HTTP_TRANSPORT)
+          throws IOException {
+    // Load client secrets.
+    InputStream in = new FileInputStream(CREDENTIALS_FILE_PATH);
+    GoogleClientSecrets clientSecrets =
+            GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+
+    // Build flow and trigger user authorization request.
+    GoogleAuthorizationCodeFlow flow =
+            new GoogleAuthorizationCodeFlow.Builder(
+                    HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES)
+                    .setDataStoreFactory(
+                            new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+                    .setAccessType("offline")
+                    .build();
+    LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(8000).build();
+    return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+  }
+
+  /**
+   * Create a MimeMessage using the parameters provided.
+   *
+   * @param toEmail email address of the receiver
+   * @param subject subject of the email
+   * @param bodyText body text of the email
+   * @return the MimeMessage to be used to send email
+   * @throws MessagingException for other failures
+   */
+  public static MimeMessage createEmail(String toEmail, String subject, String bodyText)
+          throws MessagingException {
+
+    Properties props = new Properties();
+    Session session = Session.getDefaultInstance(props, null);
```

Comment:
/* param= */ for the null argument

Reply:
Added the parameter comment of authenticator just to be more precise .